Sources/SignalRClient/HubConnection.swift

```diff
@@ -432,6 +440,47 @@ public class HubConnection {
     fileprivate func connectionDidReconnect() {
         initiateHandshake()
     }
+
+    private func resetKeepAlive() {
+        if connection.inherentKeepAlive {
+            return
+        }
+
+        logger.log(logLevel: .debug, message: "Reset keep alive")
+        keepAlivePingTask?.cancel()
+
+        keepAlivePingTask = DispatchWorkItem { self.sendKeepAlivePing() }
+
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + keepAliveIntervalInSeconds, execute: keepAlivePingTask!)
```

Comment:
Ideally we should not use the main queue for this.

Reply:
This is racey and can cause crashes. I am thinking abut a scenario where a new DispatchWorkItem is created but then there is a context switch and `sendKeepAlivePing` method executes and sets `keepAlivePingTask` to `nil`. Then the first thread resumes and tries to schedule `nil` as a task which will segfault. 
I am not sure if it is strictly necessary to ensure that these two functions can never run  concurrently but it would probably be much safer.

Reply:
I removed the force unwrap, I think that should fix the issue you mentioned. Or did you think about a different solution?